src/main/java/io/github/ImpactDevelopment/installer/target/targets/Vanilla.java

```diff
@@ -112,40 +127,23 @@ private void populateLibraries(JsonObject object, boolean multimc) {
         for (ILibrary lib : version.resolveLibraries(config)) {
             populateLib(lib, libraries);
         }
+        if (optifine != null) {
+            JsonObject lib = new JsonObject();
+            lib.addProperty("name", optifine.getOptiFineID());
+            libraries.add(lib);
+        }
+
         if (multimc) {
             object.add("+libraries", libraries);
         } else {
             object.add("libraries", libraries);
         }
-
-        populateOptifine(libraries);
-    }
-
-    private void populateOptifine(JsonArray libraries) {
-        optifineVersion().ifPresent(optifine -> {
-            JsonObject opti = new JsonObject();
-            opti.addProperty("name", "optifine:OptiFine:" + optifine);
-            libraries.add(opti);
-        });
-    }
-
-    private Optional<String> optifineVersion() {
-        return Optional.ofNullable(config.getSettingValue(OptiFineSetting.INSTANCE)).filter(optifine -> !optifine.equals(OptiFineSetting.NONE)).filter(optifine -> !optifine.equals(OptiFineSetting.MISSING));
-    }
-
-    private Optional<String> prettifiedOptifineVersion() {
-        return optifineVersion().map(str -> {
-            if (!str.startsWith(version.mcVersion + "_")) {
-                throw new IllegalStateException(str + " " + version.mcVersion);
-            }
-            return "-OptiFine" + str.substring(version.mcVersion.length());
-        });
     }
 
     private void populateLib(ILibrary lib, JsonArray libraries) {
-        if (version.mcVersion.compareTo("1.14.4") >= 0 && optifineVersion().isPresent() && lib.getName().equals("net.minecraft:launchwrapper:1.12")) {
+        if (optifine != null && optifine.getLaunchwrapperID() != null && lib.getName().equals("net.minecraft:launchwrapper:1.12")) {
             JsonObject optiLaunchWrapper = new JsonObject();
-            optiLaunchWrapper.addProperty("name", "optifine:launchwrapper-of:2.1");
+            optiLaunchWrapper.addProperty("name", optifine.getLaunchwrapperID());
```

Comment:
This is a VERY GOOD change. I almost would want this change even if this was the ONLY benefit lol.